Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -385,15 +395,53 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable {
             int firstByteIndex = _start + index * _itemsize * _step;
             object result = packBytes(format, getByteRange(firstByteIndex, _itemsize), 0, (int)_buffer.ItemSize);
 
-            // TODO: BigInteger should also be merged into an integer once the int/long merge occurs
-            if (result is BigInteger || result is double || result is float || result is Bytes) {
+            // TODO: BigInteger/long/ulong should also be merged into an integer once the int/long merge occurs
+           if (result is BigInteger || result is double || result is float || result is Bytes) {
                 return result;
+            } else if (result is long || result is ulong) {
+                return Converter.ConvertToBigInteger(result);
             } else {
                 return Convert.ToInt32(result);
             }
         }
 
         private void setAtFlatIndex(int index, object value) {
+            switch (format) {
```

Comment:
Won't this cause issues with the `_matchesBuffer` below? For example underlying buffer is a byte array but you convert to a long and call `_buffer.SetItem`.

Reply:
Each of the buffers handle overflows but they each do so differently; maybe MemoryView needs to do its own overflow check as well.

Reply:
Ah I see, it basically ends up calling the setter on the type which does its own error checking.

If you look at the CPython example below, based on the error message, it seems like the type checking is done at the memoryview level:
```
>>> m = memoryview(bytearray(b"aaa"))
>>> m[0] = 256
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: memoryview: invalid value for format 'B'
```

Seems to me like the `IBufferProtocol` interface is a bit too high level. Would be nice to have lower level access (like Span/Memory) to avoid all the overhead. Anyway, that's a problem for another time...

Reply:
I added an overflow check in MemoryView so that should be hit first, instead of the `_buffer`'s type check.